projects/valtimo/components/src/lib/components/form-io/documenten-api-uploader/documenten-api-uploader.component.ts

```diff
@@ -0,0 +1,119 @@
+import {DocumentenApiMetadata} from '../../../models';
+import {take, tap} from 'rxjs/operators';
+
+@Component({
+  selector: 'valtimo-documenten-api-formio-uploader',
+  templateUrl: './documenten-api-uploader.component.html',
+  styleUrls: ['./documenten-api-uploader.component.scss'],
+})
+export class DocumentenApiUploaderComponent implements FormioCustomComponent<Array<ResourceFile>> {
+  @Input() disabled: boolean;
+  @Input() title: string;
+  @Input() hideTitle: boolean;
+  @Input() subtitle: string;
+  @Input() maxFileSize: number;
+  @Input() hideMaxFileSize: boolean;
+  @Input() camera: boolean;
+
+  @Input() documentTitle: string;
+  @Input() disableDocumentTitle: boolean;
+  @Input() filename: string;
+  @Input() disableFilename: boolean;
+  @Input() author: string;
+  @Input() disableAuthor: boolean;
+  @Input() status: string;
+  @Input() disableStatus: boolean;
+  @Input() language: string;
+  @Input() disableLanguage: boolean;
+  @Input() documentType: string;
+  @Input() disableDocumentType: boolean;
+  @Input() description: string;
+  @Input() disableDescription: boolean;
+  @Input() confidentialityLevel: string;
+  @Input() disableConfidentialityLevel: boolean;
+
+  @Output() valueChange = new EventEmitter<Array<ResourceFile>>();
+
+  readonly uploading$ = new BehaviorSubject<boolean>(false);
+  readonly fileToBeUploaded$ = new BehaviorSubject<File | null>(null);
+  readonly modalDisabled$ = new BehaviorSubject<boolean>(false);
+  readonly showModal$ = new Subject<null>();
+  readonly hideModal$ = new Subject<null>();
+
+  constructor(
+    private readonly uploadProviderService: UploadProviderService,
+    private readonly stateService: FormIoStateService,
+    private readonly domService: FormIoDomService,
+    private readonly downloadService: DownloadService
+  ) {}
+
+  _value: Array<ResourceFile> = [];
+
+  public get value(): Array<ResourceFile> {
+    return this._value;
+  }
+
+  @Input()
+  public set value(value: Array<ResourceFile>) {
+    if (Array.isArray(value)) {
+      this._value = value;
+    }
+  }
+
+  downloadFile(resourceFile: ResourceFile) {
+    this.uploadProviderService
+      .getResource(resourceFile.data.resourceId)
+      .subscribe((resource: ResourceDto) => {
+        this.downloadService.downloadFile(resource.url, resource.resource.name);
+      });
+  }
+
+  fileSelected(file: File) {
+    this.fileToBeUploaded$.next(file);
+    this.showModal$.next(null);
+  }
+
+  deleteFile(resourceId: string): void {
+    this.domService.toggleSubmitButton(true);
+    this._value = this._value.filter((file: ResourceFile) =>
+      file?.data ? file?.data?.resourceId !== resourceId : true
+    );
+    this.valueChange.emit(this._value);
+  }
+
+  metadataSet(metadata: DocumentenApiMetadata): void {
+    this.uploading$.next(true);
+    this.hideModal$.next(null);
+
+    this.fileToBeUploaded$
+      .pipe(
+        take(1),
+        tap(file => {
+          console.log('upload', file, metadata);
```

Comment:
check the need for this console.log and maybe remove/replace it

Reply:
Will be removed once BE endpoints are ready